src/azul/deployment.py

```diff
@@ -283,5 +295,174 @@ def _sanitize_tf(tf_config: JSON) -> JSON:
     return {k: v for k, v in tf_config.items() if v}
 
 
+def _normalize_tf(tf_config: Union[JSON, JSONs]) -> Iterable[Tuple[str, AnyJSON]]:
+    """
+    Certain levels of a Terraform JSON structure can either be a single
+    dictionary or a list of dictionaries. For example, these are equivalent:
+
+        {"resource": {"resource_type": {"resource_id": {"foo": ...}}}}
+        {"resource": [{"resource_type": {"resource_id": {"foo": ...}}}]}
+
+    So are these:
+
+        {"resource": {"type": {"id": {"foo": ...}, "id2": {"bar": ...}}}}
+        {"resource": {"type": [{"id": {"foo": ...}}, {"id2": {"bar": ...}}]}}
+
+    This function normalizes input to prefer the second form of both cases to
+    make parsing Terraform configuration simpler. It returns an iterator of the
+    dictionary entries in the argument, regardless which form is used.
+
+    >>> list(_normalize_tf({'foo': 'bar'}))
+    [('foo', 'bar')]
+
+    >>> list(_normalize_tf([{'foo': 'bar'}]))
+    [('foo', 'bar')]
+
+    >>> list(_normalize_tf({"foo": "bar", "baz": "qux"}))
+    [('foo', 'bar'), ('baz', 'qux')]
+
+    >>> list(_normalize_tf([{"foo": "bar"}, {"baz": "qux"}]))
+    [('foo', 'bar'), ('baz', 'qux')]
+
+    >>> list(_normalize_tf([{"foo": "bar", "baz": "qux"}]))
+    [('foo', 'bar'), ('baz', 'qux')]
+    """
+    if isinstance(tf_config, dict):
+        return tf_config.items()
+    elif isinstance(tf_config, list):
+        return chain.from_iterable(d.items() for d in tf_config)
+    else:
+        assert False, type(tf_config)
+
+
+def populate_tags(tf_config: JSON) -> JSON:
+    """
+    Add tags to all taggable resources and change the `name` tag to `Name`
+    for tagged AWS resources.
+
+    >>> from azul.doctests import assert_json
+    >>> assert_json(\
```

Comment:
Is using `\` to make this span several lines acceptable? Without it, the code would exceed the 120-character limit.

Alternatively, I could extract the argument to `populate_tags` as a variable?